src/templates/next/components/shared/CollectionSort/CollectionSort.tsx

```diff
@@ -0,0 +1,95 @@
+import { Paragraph } from "../../../typography/Paragraph"
+import { useState } from "react"
+import { MdKeyboardArrowDown } from "react-icons/md"
+import CollectionSortProps, {
+  SortDirection,
+  SortKey,
+} from "~/templates/next/types/CollectionSort"
+import { BiCheck } from "react-icons/bi"
+
+interface SortConfig {
+  sortBy: SortKey
+  sortDirection: SortDirection
+}
+
+const SortOptions = ["Most recent", "Least recent"] as const
+
+type SortOption = (typeof SortOptions)[number]
+
+const SortOptionToConfigMap: Record<SortOption, SortConfig> = {
+  "Most recent": {
+    sortBy: "date",
+    sortDirection: "desc",
+  },
+  "Least recent": {
+    sortBy: "date",
+    sortDirection: "asc",
+  },
+}
+
+const SortConfigToOptionMap: Record<
+  SortKey,
+  Record<SortDirection, SortOption>
+> = {
+  date: {
+    desc: "Most recent",
+    asc: "Least recent",
+  },
+}
+
+const CollectionSort = ({
+  sortBy,
+  setSortBy,
+  sortDirection,
+  setSortDirection,
+}: CollectionSortProps) => {
+  const [showSortOptions, setShowSortOptions] = useState(false)
+  const selectedSortOption = SortConfigToOptionMap[sortBy][sortDirection]
+  return (
+    <div className="relative">
+      <div className="flex flex-col gap-2">
+        <p className={`${Paragraph[2]} text-content-strong`}>Sort by</p>
+        <button
+          className={`flex gap-6 justify-between border border-divider-medium py-2.5 px-4 rounded-[4px] ${
+            showSortOptions && "border-2 border-focus-outline"
+          }`}
+          aria-label={
+            showSortOptions ? "Hide sort options" : "Show sort options"
+          }
+          onClick={() => setShowSortOptions(!showSortOptions)}
+        >
+          {selectedSortOption}
+          <MdKeyboardArrowDown
+            className={`w-5 h-auto flex-shrink-0 transition-all duration-300 ease-in-out ${
+              showSortOptions ? "rotate-180" : "rotate-0"
+            }`}
+          />
+        </button>
+      </div>
+      {showSortOptions && (
+        <div className="absolute w-full bg-white flex flex-col gap-0 rounded-[4px] py-3">
+          {SortOptions.map((option) => (
+            <button
+              key={option}
+              className={`flex justify-between items-center w-full rounded-[4px] py-2.5 px-4 text-left hover:bg-interaction-main-subtle-hover ${
+                option === selectedSortOption && "border-2 border-focus-outline"
```

Comment:
nit: more of a design feedback but I feel having the border is a little weird, especially since the original dropdown also has a border, which feels like seeing double.

<img width="480" alt="Screenshot 2024-04-01 at 10 21 50" src="https://github.com/isomerpages/isomer-components/assets/27919917/d3106a56-85e6-4a3f-b04f-8dbb053dc31f">

Reply:
okay, will bring up during design review